app/javascript/components/widget-transition-controls.jsx

```diff
@@ -75,7 +75,7 @@ const WidgetTransitionControls = ({
   const widgetTransitionTime = current.durationSeconds * 1000;
 
   const [state, dispatch] = useReducer(reducer, {
-    startTimestamp: Date.now(),
+    startTimestamp: null,
```

Comment:
What bug is this fixing?

Reply:
startTimestamp's value isn't meant to be set in this reducer (and wasn't until a recent refactor PR which accidentally changed it). These changes produced a bug (exposed by implementing startTimer) which could cause widgets to skip over themselves.

Reply:
This would be good to include in the commit message.  A format like this:

```
Main Description

Detail for a review that would be helpful for them to understand why you making the changes you are are without them having to run it and recreate.
```

Reply:
This does look a lot better. I think I'm going to write a lot more commits like this going forward. Thanks, Mike